### PR TITLE
MINIFICPP-1994 - Route flowfiles with invalid SQL to Failure

### PR DESCRIPTION
**Pull request by adamdebreceni**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by lordgamez — APPROVED** [review]

File: libminifi/test/TestBase.cpp [hunk 1]
```diff
@@ -564,7 +588,7 @@ std::unique_ptr<minifi::Connection> TestPlan::buildFinalConnection(const std::sh
   std::stringstream connection_name;
   connection_name << processor->getUUIDStr() << "-to-" << processor->getUUIDStr();
   auto connection = std::make_unique<minifi::Connection>(flow_repo_, content_repo_, connection_name.str());
-  connection->addRelationship(termination_);
+  connection->addRelationship(termination_.value());
```
Comment (szaszm) [inline libminifi/test/TestBase.cpp:591]: Shouldn't we check if `termination_` is set to something?

File: extensions/sql/processors/ExecuteSQL.h [hunk 2]
```diff
@@ -58,6 +60,9 @@ class ExecuteSQL : public SQLProcessor, public FlowFileSource {
 
   EXTENSIONAPI static const std::string RESULT_ROW_COUNT;
   EXTENSIONAPI static const std::string INPUT_FLOW_FILE_UUID;
+
+ private:
+  static const std::shared_ptr<core::logging::Logger> logger_;
```
[the remark this answers is not in the document]
Comment (adamdebreceni) [reply extensions/sql/processors/ExecuteSQL.h]: removed it

File: libminifi/test/TestBase.cpp (hunk 1 above)
Comment (adamdebreceni) [reply libminifi/test/TestBase.cpp:591]: in theory `buildFinalConnection` is only called when `termination_` is set, added `gsl_Expects` to verify this precondition

File: libminifi/test/TestBase.cpp [hunk 3]
```diff
@@ -234,12 +234,16 @@ std::shared_ptr<minifi::core::Processor> TestPlan::addProcessor(const std::share
   processor->setFlowIdentifier(flow_version_->getFlowIdentifier());
   processor_mapping_[processor->getUUID()] = processor;
   if (!linkToPrevious) {
-    termination_ = *(relationships.begin());
+    if (!std::empty(relationships)) {
+      termination_ = *(relationships.begin());
```
Comment (szaszm) [inline libminifi/test/TestBase.cpp:238]:
```suggestion
      termination_ = relationships.front();
```

File: libminifi/test/TestBase.cpp [hunk 4]
```diff
@@ -234,12 +234,16 @@ std::shared_ptr<minifi::core::Processor> TestPlan::addProcessor(const std::share
...
+    }
   } else {
     std::shared_ptr<minifi::core::Processor> last = processor_queue_.back();
     if (last == nullptr) {
       last = processor;
-      termination_ = *(relationships.begin());
+      if (!std::empty(relationships)) {
+        termination_ = *(relationships.begin());
```
Comment (szaszm) [inline libminifi/test/TestBase.cpp:245]:
```suggestion
        termination_ = relationships.front();
```

File: extensions/sql/processors/PutSQL.h [hunk 5]
```diff
@@ -53,6 +55,9 @@ class PutSQL : public SQLProcessor {
   void processOnTrigger(core::ProcessContext& context, core::ProcessSession& session) override;
 
   void initialize() override;
+
+ private:
+  static std::shared_ptr<core::logging::Logger> logger_;
```
[the remark this answers is not in the document]
Comment (adamdebreceni) [reply extensions/sql/processors/PutSQL.h]: removed

File: libminifi/test/TestBase.cpp (hunk 3 above)
Comment (szaszm) [reply libminifi/test/TestBase.cpp]: It seems to be breaking the build. I thought this is a vector, but the error msg says it's an `std::initializer_list`.

Comment (adamdebreceni) [reply libminifi/test/TestBase.cpp:238]: reverted

**Review by szaszm — APPROVED** [review]

File: extensions/sql/processors/ExecuteSQL.h [hunk 6]
```diff
@@ -42,7 +43,8 @@ class ExecuteSQL : public SQLProcessor, public FlowFileSource {
   }
 
   EXTENSIONAPI static const core::Relationship Success;
-  static auto relationships() { return std::array{Success}; }
+  EXTENSIONAPI static const core::Relationship Failure;
+  static auto relationships() { return std::array{Success, Failure}; }
```
Comment (fgerlits) [inline extensions/sql/processors/ExecuteSQL.h:47]: please add these two new `failure` relationships to `PROCESSORS.md`, too

Comment (adamdebreceni) [reply extensions/sql/processors/ExecuteSQL.h:47]: added

**Review by fgerlits — APPROVED** [review]

**Review by martinzink — APPROVED** [review]

**Pull request closed without merging**